checkov/terraform/graph_builder/local_graph.py

```diff
@@ -89,9 +90,34 @@ def build_graph(self, render_variables: bool) -> None:
                 edges_count = len(self.edges)
                 self._build_cross_variable_edges()
                 logging.info(f"Found {len(self.edges) - edges_count} cross variable edges")
+            self._remove_empty_foreach_edges()
         else:
             self.update_vertices_fields()
 
+    def _remove_empty_foreach_edges(self) -> None:
```

Comment:
shouldn't we also remove the resource itself it's has count=0 or for_each={}?

Reply:
I thought about it but didn't know if that's how we want it to behave.
Currently I handled the edges to remove any non-existent connections.